packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1710,15 +1709,14 @@ export class ContainerRuntime
 		this.updateDocumentDirtyState(newState);
 	}
 
-	private async applyStashedOp(
-		type: ContainerMessageType,
-		op: ISequencedDocumentMessage,
-	): Promise<unknown> {
-		switch (type) {
+	private async applyStashedOp(op: string): Promise<unknown> {
+		const { type, contents } = JSON.parse(op);
```

Comment:
This parsing is also done below. Do we need to guard against messages with different schemas here? Is there a possibility of getting a badly structured message here?

Reply:
It's not currently possible for this to be malformed because the `PendingStateManager` does conversion on it. However, in future version it will no longer do the conversion, so it probably makes sense to have some asserts verifying the structure of the content (currently determine by `ContainerRuntime.submit(...)`).